### DISCUSSION TITLE
transmitDirect()

### DISCUSSION THREAD

**Discussion by Crsarmv7l** [jgromes/RadioLib#1060] [Q&A] [answered]
I have been browsing the discussions and code for a bit regarding transmitDirect. I see it used in pager and a few other protocols, but I haven't seen much other documentation regarding it.

I did find this discussion from a few years ago https://github.com/jgromes/RadioLib/discussions/312 that transmitDirect only generates unmodulated carriers.

Is that still the case?

My reasons for seeking to use it are as follows:
- I already have RX set up for direct mode (it would keep things consistent)
- Using direct mode for TX would eliminate FIFO buffer limitations
- Per Ti Forums you *cannot* disable preamble/syncword and still use the FIFO's with the CC1101

That last one is the biggie.

**Comment by Velocet** [comment]
I can only speak for Si446x and SX127x:
Since you have an unmodulated carrier you connect CLK and DATA to the MCUs UART. These signals then drive the modulation and give you complete freedom over how you send your packet.

**Reply by Crsarmv7l** [reply]
The CC1101 provides clock on GDO0 in direct mode, which is why that discussion strikes me as odd.

**Comment by jgromes** [comment, marked as answer]
>  transmitDirect only generates unmodulated carrier

That is correct, when you enable transmitDirect on any module in RadioLib that supports direct mode, the module will start transmitting unmodulated carrier at whatever frequency and power is currently set. However, you can then start modulating the signal, either by changing the configured freuquency (for low-speed protocols, like RTTY), or by applying some waveforms to the modules DIO/GDO pins.
